mapper/pojo/src/main/java/org/hibernate/search/mapper/pojo/mapping/impl/PojoWorkPlanImpl.java

```diff
@@ -120,6 +122,21 @@ public void prepare() {
 		}
 	}
 
+	@Override
+	public <E> Set<Class<? extends E>> getIndexedTypesPolymorphic(Class<E> entityType) {
```

Comment:
This has nothing to do with the work plan. Rule of thumb: work plans should only provide operations related to... working on the index.

The `getIndexedTypesPolymorphic` method should be exposed in `PojoMappingDelegate` and `HibernateOrmMapping`. You can already see metadata-querying methods in there: `isWorkable`, `isIndexable`, ...

`PojoMappingDelegateImpl` would have the actual implementation, and `HibernateOrmMappingImpl` would just delegate.

In `MassIndexerImpl`, this should not be a problem: you have a `mapping` attribute, so you can just use it instead of the work plan.